sandbox/libs/authn/src/main/java/org/opensearch/authn/Permission.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.authn;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class Permission {
+    private final String[] permissionChunks; 
+
+    public Permission(final String permission) {
+        this.permissionChunks = permission.split("\\.");
+    }
+
+    public boolean matches(final String permissionRequired) {
```

Comment:
seems like we can leverage `equals` method here. Thoughts?

Reply:
A user might have a grant represented as `opensearch.indexing.index`, and the checked permission will be `opensearch.indexing.index.create`.  This isn't an equality check, but a check if the permission grant matches the permission checked.

As I'm writing this out, it might make sense to create different objects to represent the grant vs the permission them selves - I'll revisit this